src/librustc_mir/interpret/snapshot.rs

```diff
@@ -358,6 +356,21 @@ impl<'a, 'mir, 'tcx, Ctx> Snapshot<'a, Ctx> for &'a Frame<'mir, 'tcx>
     }
 }
 
+impl<'a, 'tcx, Ctx> Snapshot<'a, Ctx> for &'a LocalState<'tcx>
+    where Ctx: SnapshotContext<'a>,
+{
+    type Item = LocalValue<(), AllocIdSnapshot<'a>>;
+
+    fn snapshot(&self, ctx: &'a Ctx) -> Self::Item {
+        self.state.snapshot(ctx)
```

Comment:
We otherwise follow the pattern to use `LocalState { state, layout: _ } = self` to make sure we revisit this when new fields get added.

`impl_snapshot_for!` doesn't work?

Reply:
nope, it's a very simple macro with no support for lifetimes or so

Reply:
Done and opened https://github.com/rust-lang/rust-clippy/issues/3724 to make sure we take care of this issue once and for all at some point